conda_lock.py

```diff
@@ -16,31 +16,33 @@
 import yaml
 import os
 import requests
+import tempfile
+
 
 DEFAULT_PLATFORMS = ["osx-64", "linux-64", "win-64"]
-FAKE_PREFIX_NAME = "__magicmarker"
-FAKE_PKGS_ROOT = "/tmp/something/that/does/not/exist"
 
 
 def solve_specs_for_arch(channels, specs, platform):
     # type: (typing.List[str], typing.List[str], str) -> dict
-    args = [
-        "conda",
-        "create",
-        "-p",
-        FAKE_PREFIX_NAME,
-        "--override-channels",
-        "--dry-run",
-        "--json",
-    ]
-    for channel in channels:
-        args.extend(["-c", channel])
-    args.extend(specs)
 
     env = dict(os.environ)
-    env.update({"CONDA_SUBDIR": platform, "CONDA_PKGS_DIRS": FAKE_PKGS_ROOT})
-    json_output = subprocess.check_output(args, env=env)
-    # print(json_output)
+    with tempfile.TemporaryDirectory() as CONDA_PKGS_DIRS:
+        env.update({"CONDA_SUBDIR": platform, "CONDA_PKGS_DIRS": CONDA_PKGS_DIRS})
+
+        args = [
+            "conda",
+            "create",
+            "--prefix",
+            f"{CONDA_PKGS_DIRS}_prefix",
```

Comment:
@mariusvniekerk, after asking around I believe this is safer b/c, if one day dry-run does the right thing and fails with an existing name, we would have a broken conda-lock.

Reply:
```suggestion
            f"{CONDA_PKGS_DIRS}/prefix",
```

Reply:
or maybe with pathlib's join